wordpress-mobile/WordPress-Android
# Fix reader related posts loading the wrong post after rotation

Fixes #14195 

### Issue description
There's an issue that when using the reader to see Related Posts, after going more than 2 levels deep in related posts and rotating the device, the post loaded will be the first one displayed.

### Work done
I changed where the Fragment restores its state from `onActivityCreated` to `onViewCreated`.
`onActivityCreated` is called after `onViewCreated`, so the state was being restored after the view was ready and the post was already loaded with the wrong id.

### To test
1. Open a wp.com post on Reader tab.
2. Scroll to bottom of the post details page.
3. Notice that related posts are displayed.
4. Click a related post.
5. Notice that selected related post details are shown on "Related Post" screen with a cross button.
6. Open another related post from the previous "Related Post" screen.
7. Notice that selected related post details are shown on "Related Post" screen with a cross button.
8. Rotate the device.
9. Notice that first related post is shown instead of the last one.

### Video
<img src="https://user-images.githubusercontent.com/628412/133923158-9af3d32e-59a0-4dcd-83d0-6aaffc72fdda.gif" width="300" />

## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
None

3. What automated tests I added (or what prevented me from doing so)
None. No automated tests are present for the area (just adding them should be a separate PR)

PR submission checklist:
